eng/scripts/New-VcpkgPortDefinition.ps1

```diff
@@ -0,0 +1,57 @@
+<#
+.SYNOPSIS
+Replaces token content in each file of the SourceDirectory (assumes no subfolders)
+
+.PARAMETER SourceDirectory
+Location of vcpkg port files
+(usually `<artifact-path>/packages/<package-name>/vcpkg/port`)
+
+.PARAMETER Version
+Replaces %VERSION% token in files with the value of this parameter
+
+.PARAMETER Url
+Replaces %URL% token in files with the value of this parameter
+
+.PARAMETER Filename
+Replaces %FILENAME% token in files with the value of this parameter
+
+.PARAMETER Sha512
+Replaces %SHA512% token in files with the value of this parameter
+
+#>
+
+
+param (
+    [Parameter(Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [string] $SourceDirectory,
+
+    [Parameter(Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [string] $Version,
+
+    [Parameter(Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [string] $Url,
+
+    [Parameter(Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [string] $Filename,
+
+    [Parameter(Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [string] $Sha512
+)
+
+$files = Get-ChildItem -Path $SourceDirectory
+
+foreach ($file in $files) {
```

Comment:
Is this actually useful as a separate script or should we just add it to the initialize script?

Reply:
This decouples the operation of setting up the vcpkg template files from the destructive operation of creating a tag. It's useful if you're iterating on token substitution locally or want to test build outputs with a local vcpkg before opening a PR against the main vcpkg repo.